editor/src/core/model/element-metadata-utils.ts

```diff
@@ -2291,6 +2292,30 @@ function fillMissingDataFromAncestors(mergedMetadata: ElementInstanceMetadataMap
     }
   })
 
+  // conditionals should receive specialSizeMeasurements.display from their parent
+  const conditionals = Object.keys(workingElements).filter((p) =>
+    MetadataUtils.isConditionalFromMetadata(workingElements[p]),
+  )
+
+  fastForEach(conditionals, (pathStr) => {
+    const elem = workingElements[pathStr]
+
+    const parentPathStr = EP.toString(EP.parentPath(EP.fromString(pathStr)))
+
+    const parentSpecialSizeMeasurements = workingElements[parentPathStr]?.specialSizeMeasurements
+
+    if (parentSpecialSizeMeasurements != null) {
+      workingElements[pathStr] = {
+        ...elem,
+        specialSizeMeasurements: {
+          ...elem.specialSizeMeasurements,
+          display: parentSpecialSizeMeasurements.display,
+        },
+      }
+    }
+  })
```

Comment:
I think the structure would be way more apparent here if each of these passes had their separate functions, which take `workingElements`, do their thing, and return an updated `workingElements`

Reply:
I absolutely agree, I just did not want to give up on the mutation of `workingElements` for performance reasons, maybe it is an overoptimization though.